src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -280,8 +280,16 @@ export class SearchSource {
 
     let response;
     if (getConfig(UI_SETTINGS.COURIER_BATCH_SEARCHES)) {
+      /**
+       * batch search will issue `_msearch` requests to OpenSearch, which is not likely to work with
+       *  multiple data sources. maybe deprecate this, or simply disable it when multiple data source is supported?
+       * zengyan NOTE: Kibana is going to remove it: https://github.com/elastic/kibana/issues/55140
+       */
```

Comment:
let's remove this comment in code and create an issue to track the _msearch support

Reply:
create one here https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2174